systems/roadbuilding.go

```diff
@@ -0,0 +1,414 @@
+package systems
+
+import (
+	"fmt"
+	"image/color"
+	"time"
+
+	"engo.io/ecs"
+	"engo.io/engo"
+	"engo.io/engo/common"
+	"github.com/EngoEngine/TrafficDefense/systems/ui"
+	"github.com/luxengine/math"
+)
+
+const (
+	roadZIndex = -1
+
+	costPerUnit float32 = 100
+	laneWidth   float32 = 10
+)
+
+var (
```

Comment:
Only builtin types can be `const`, so all `color.RGBA` have to be `var`s. (limitation of Go)